voyager-core/src/androidMain/kotlin/cafe/adriel/voyager/androidx/AndroidScreenLifecycleOwner.kt

```diff
@@ -187,6 +191,20 @@ public class AndroidScreenLifecycleOwner private constructor() :
         }
     }
 
+    private fun registerContextResetCallback(context: Context) {
+        if (!isContextResetCallbackRegistered) {
+            val lifecycleOwner = (context as? LifecycleOwner) ?: return
+            lifecycleOwner.lifecycle.addObserver(
+                object : DefaultLifecycleObserver {
+                    override fun onDestroy(owner: LifecycleOwner) {
+                        atomicContext.set(null)
+                        isContextResetCallbackRegistered = false
+                    }
+                }
+            )
+        }
+    }
```

Comment:
Does this actual need to be on the `lifecycleOwner.lifecycle`? This lifecycle owner is from Voyager, it emulates the events based on Voyager navigation events, so this `onDestroy` is getting called when the `onDispose(screen)` is called.

Maybe we can move this `atomicContext.set(null)` to the `onDispose` ?

But I still not convinced that this would actual work honestly. For example, the `onDispose` I don't know is called before the screen is completed removed from the composition, for example with ScreenTransitions, does the `getHooks()` get called after `onDispose` is called? Can you validate that?
If this is the case, we could wrap the `atomicContext.compareAndSet(null, context)` in a if statement validating if the current `lifecycle state` is different from DESTROYED.

cc @Syer10